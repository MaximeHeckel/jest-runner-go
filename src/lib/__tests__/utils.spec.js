import {
  parseGoOutput,
  toTestResult,
} from '../utils';


const path = 'github.com/MaximeHeckel/go-test-runner/example/stringutil';
const start = 1510292283895;
const output = [ 'ok  \tgithub.com/MaximeHeckel/go-test-runner/example/stringutil\t0.007s','' ];

const test = {
  passed: 1,
  failed: 0,
  failureMessage: '',
  duration: 336,
  end: 1510292594284,
  name: 'github.com/MaximeHeckel/go-test-runner/example/stringutil2',
}

describe('parseGoOutput', () => {
  it('parses the golang test output accordingly', () => {
    expect(parseGoOutput(path, start, output)).toMatchSnapshot();
  });
});

describe('toTestResult', () => {
  it('returns a properly formatted test result', () => {
    expect(toTestResult(test)).toMatchSnapshot();
  });
});