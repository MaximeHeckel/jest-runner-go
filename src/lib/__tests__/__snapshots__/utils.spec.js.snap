// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseGoOutput parses the golang test output accordingly 1`] = `
Object {
  "duration": 0,
  "end": 0,
  "failed": 0,
  "failureMessage": "",
  "name": "github.com/MaximeHeckel/go-test-runner/example/stringutil",
  "passed": 1,
}
`;

exports[`toTestResult returns a properly formatted test result 1`] = `
Object {
  "ancestorTitles": Array [],
  "duration": 336,
  "failureMessages": "",
  "fullName": "github.com/MaximeHeckel/go-test-runner/example/stringutil2",
  "numPassingAsserts": 1,
  "status": "passed",
  "title": "github.com/MaximeHeckel/go-test-runner/example/stringutil2",
}
`;
